### PR TITLE
Storybook entries for account management pages

### LINKED ISSUE #368: Account management frontend
Implement these rpcs

```
rpc ResetPassword(ResetPasswordReq) returns (google.protobuf.Empty) {
    // Sends a forgot password email to the given user if the user exists, returns no output (so you can't go around guessing email addresses)
  }

  rpc CompletePasswordReset(CompletePasswordResetReq) returns (google.protobuf.Empty) {
    // Triggered when the user goes to the link sent in the forgot password email
  }

  rpc CompleteChangeEmail(CompleteChangeEmailReq) returns (google.protobuf.Empty) {
    // When a user opens a changed email confirmation, the frontend should call this function with the token
  }
```

To-do list:
- [x] ResetPassword
- [x] CompletePasswordReset
- [x] CompleteChangeEmail
- [x] ChangePassword
  - [x] Improve UX by checking if user has password with GetAccountInfo
- [x] ChangeEmail

Edit: and below

### PR DESCRIPTION
**Pull request by darrenvong**
Completes and closes #368... should have added this in my last PR rather than waiting for the big tidy up first lol, but I guess that helps give some structure for where to put this

### CODE REVIEW

**Review by lucaslcode — CHANGES_REQUESTED** [review]
Great, much neater than how I was doing it. I will copy it.

Just something I didn't pick up on before, the message shouldn't say "Leave the fields blank to unset your password" if the user doesn't have a password already.

File: app/frontend/src/features/auth/password/ChangePasswordPage.tsx [hunk 1]
```diff
@@ -82,8 +82,10 @@ export default function ChangePasswordPage() {
             </Alert>
           )}
           <Typography className={classes.infoText} variant="body1">
-            Please enter a new password, or leave the "New password" and
-            "Confirm password" fields blank to unset your password.
+            Please enter a new password,{" "}
+            {accountInfo?.hasPassword
+              ? 'or leave the "New password" and "Confirm password" fields blank to unset your password.'
+              : "or leave the fields blank to unset your password."}
```
Comment (lucaslcode) [inline app/frontend/src/features/auth/password/ChangePasswordPage.tsx]: Still the same problem, no? The user has no password and they see `Please enter a new password, or leave the fields blank to unset your password.` But they already don't have a password.

Comment (darrenvong) [reply app/frontend/src/features/auth/password/ChangePasswordPage.tsx]: Oh yeah what was I thinking 🤦

File: app/frontend/src/features/auth/password/ChangePasswordPage.tsx [hunk 2]
```diff
@@ -82,8 +82,9 @@ export default function ChangePasswordPage() {
             </Alert>
           )}
           <Typography className={classes.infoText} variant="body1">
-            Please enter a new password, or leave the "New password" and
-            "Confirm password" fields blank to unset your password.
+            {accountInfo?.hasPassword
+              ? 'Please enter a password, or leave the "New password" and "Confirm password" fields blank to unset your password.'
+              : "Leave the fields blank to unset your password."}
```
Comment (lucaslcode) [inline app/frontend/src/features/auth/password/ChangePasswordPage.tsx]:
```suggestion
              : "Please enter a password."}
```

File: app/frontend/src/features/auth/password/ChangePasswordPage.test.tsx [hunk 3]
```diff
@@ -129,6 +134,9 @@ describe("ChangePasswordPage", () => {
       // Wait for new password field/form to show up first, otherwise old password not visible is always
       // gonna be true
       expect(await screen.findByLabelText("New password")).toBeVisible();
+      expect(
+        screen.getByText("Leave the fields blank to unset your password.")
```
Comment (lucaslcode) [inline app/frontend/src/features/auth/password/ChangePasswordPage.test.tsx]:
```suggestion
        screen.getByText("Please enter a password")
```

**Review by lucaslcode — CHANGES_REQUESTED** [review]
😊

File: app/frontend/src/features/auth/password/ChangePasswordPage.tsx (hunk 2 above)
Comment (darrenvong) [reply app/frontend/src/features/auth/password/ChangePasswordPage.tsx]: I did think after I committed (and was too late to turn back on my laptop 💤 ) if it makes any sense to clear your password if you don't have one already 😂

**Review by lucaslcode — CHANGES_REQUESTED** [review]
Test failed. But it looks like it should have worked???

**Comment by darrenvong** [comment]
It's because it's a string value and so require precise match, but the value missed a full stop 😬

**Review by lucaslcode — APPROVED** [review]
If it's green

**Pull request merged**